### PR TITLE
Force the right git commit

### PR DESCRIPTION
**Pull request by dpoirier**
Instead of trying to pull, do a fetch to update the
remote data, then reset --hard to switch to the
commit hash or branch we want without having to
worry about merges, local branches etc.

### CODE REVIEW

File: fabulaws/library/wsgiautoscale/api.py [hunk 1]
```diff
@@ -746,13 +746,22 @@ def bootstrap(purge=False):
...
 def update_source(changeset=None):
     """Checkout the latest code from repo."""
-    if changeset is None:
-        changeset = env.branch
     require('environment', provided_by=env.environments)
     with cd(env.code_root):
         sudo('find . -name "*.pyc" -delete')
-        vcs('pull')
-        vcs('update', [changeset])
+        if env.vcs_cmd.endswith('git'):
```
Comment (vkurup) [inline fabulaws/library/wsgiautoscale/api.py]: cool, i wouldn't have thought to do this (i'm so git-centric)

**Comment by vkurup** [comment]
:+1:

File: fabulaws/library/wsgiautoscale/api.py (hunk 1 above)
Comment (dpoirier) [inline fabulaws/library/wsgiautoscale/api.py]: This hg v. git stuff is all over the 'vcs' command.  otherwise I'd have forgotten about it too.

**Comment by tobiasmcnulty** [comment]
:+1:

**Pull request merged**